docs/ghc-deprecation.md

```diff
@@ -0,0 +1,37 @@
+# GHC support deprecation policy
+
+- `haskell-language-server`(HLS) is highly tied to ghc api so much so that it needs a specific flavour for each ghc minor version to ensure it will work in a reliable way
+- It supposes the codebase is riddled with code conditioned to each supported ghc versions. It even needs entire packages to fully support older versions of ghc.
+- Our continouos integracion setup has to cover all those cases so it have to use lot of resources to test and build the executable.
+- So we need to limit the ghc support to save maintainers and contributors time and reduce ci resources consumption. It becomes vital to make HLS development manageable.
+- At same time we aim to support the right balance of ghc versions to minimize impact to final users who usually needs to keep using older ghc versions, even if they are out of the window support offered by ghc itself.
+- To establish the policy and the possible exceptions we aim to take in account:
+   - completeness of support: the GHC flavour should include all plugins and features
+   - ghc versions supported by newer [stackage](https://www.stackage.org/) LTS's
+   - ghc versions supported by default in the most popular [linux distributions](https://repology.org/project/ghc/versions)
+   - the specific history of ghc releases and their realibility on the major operating systems (linux, windows, macos)
+- It worths to note that users of deprecated ghc versions always will have the option of:
+  - Continue using the last HLS with support for their ghc version. So they will miss new bug fixes and features.
+  - For deprecated minor versions of a still supported major version, try to build HLS from source.  
+    - We will not guarentee it will work but will do quite likely.
+
```

Comment:
```suggestion
`haskell-language-server`(HLS) is highly tied to the ghc api.This imposes a high maintenance cost:
- The codebase is littered with conditional logic,
- We own auxiliary packages to support older versions of ghc.
- CI has to cover all the supported versions.

So we need to limit the ghc support to save maintainers and contributors time and reduce CI resources.

At same time we aim to support the right balance of ghc versions to minimize impact to final users.
To establish the policy we take into account:
   - completeness: support includes all plugins and features
   - The most recent [stackage](https://www.stackage.org/) LTS snapshot
   - The GHC versions used in the most popular [linux distributions](https://repology.org/project/ghc/versions)

It is worth noting that users always will have the option of:
  - using the last HLS with support for their ghc version.
  - build HLS from source.  
```

Reply:
It looks cleaner, thanks, but why does it remove `the specific history of ghc releases and their realibility...`?
I think this is the real reason behind keep 8.6.5 and 8.10.5 and it is previsible that we should take that in account, even being an ambiguous consideration

Reply:
Is it? I thought we are not making any exceptions and therefore not keeping 8.6.5, and for 8.10.5 the reason was a Linux distro.

I removed because it's poorly defined and likely to lead to disagreements, and I don't think it's needed.

Reply:
Well , we disagree here i think users will like to know we aim to be flexible and will listen their opinion about their use cases about ghc versioning. It is more a statement of principles than a definition.
Maintainers will have always the last word.